coffeecard/CoffeeCard.Library/Services/v2/AccountService.cs

```diff
@@ -289,5 +300,44 @@ await _context.Users
                     .ExecuteUpdateAsync(u => u.SetProperty(u => u.UserGroup, item.UserGroup));
             }
         }
+
+        public async Task SendMagicLinkEmail(string email, LoginType loginType)
+        {
+            var user = await _context.Users
+                .Where(u => u.Email == email)
+                .FirstOrDefaultAsync();
+            if (user is null)
+            {
+                // Should not throw error to prevent showing a malicious user if an email is already registered
+                return;
+            }
+            var magicLinkTokenHash = _tokenServiceV2.GenerateMagicLink(user);
+            await _emailServiceV2.SendMagicLink(user, magicLinkTokenHash, loginType);
+        }
+
+        public async Task<UserLoginResponse> GenerateTokenPair(string token)
```

Comment:
Hmm, in my opinion `UserLoginResponse` is a data object representing/carrying a token pair. But on the other hand, token might be a bit overloaded / generic for this use-case so I have renamed it. Hope you like the new name, otherwise please suggest alternatives :)